internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec_test.go

```diff
@@ -99,27 +98,101 @@ var _ = Describe("Validate clusterlogforwarderspec", func() {
 				Expect(clfStatus.Inputs[inputName]).To(HaveCondition("Ready", false, loggingv1.ReasonInvalid, expectedErrMsg))
 			}
 
-			checkPorts := func(port int32, format string, expectedErrMsg string) {
+			checkPortAndHTTPFormat := func(port int32, format string, expectedErrMsg string) {
 				checkReceiver(
 					&loggingv1.ReceiverSpec{
-						HTTP: &loggingv1.HTTPReceiver{
-							Port:   port,
-							Format: format,
+						Type: loggingv1.ReceiverTypeHttp,
+						ReceiverTypeSpec: &loggingv1.ReceiverTypeSpec{
+							HTTP: &loggingv1.HTTPReceiver{
+								Port:   port,
+								Format: format,
+							},
 						},
 					},
 					expectedErrMsg,
 					map[string]bool{constants.VectorName: true},
 				)
 			}
 
+			checkPortAndSyslogProtocol := func(port int32, protocol string, expectedErrMsg string) {
+				checkReceiver(
+					&loggingv1.ReceiverSpec{
+						Type: loggingv1.ReceiverTypeSyslog,
+						ReceiverTypeSpec: &loggingv1.ReceiverTypeSpec{
+							Syslog: &loggingv1.SyslogReceiver{
+								Port:     port,
+								Protocol: protocol,
+							},
+						},
+					},
+					expectedErrMsg,
+					map[string]bool{constants.VectorName: true},
+				)
+			}
+
+			checkReceiverType := func(receiverType string, expectedErrMsg string) {
+				checkReceiver(
+					&loggingv1.ReceiverSpec{
+						Type:             receiverType,
+						ReceiverTypeSpec: &loggingv1.ReceiverTypeSpec{},
+					},
+					expectedErrMsg,
+					map[string]bool{constants.VectorName: true},
+				)
+			}
+
+			for _, port := range []int32{-1, 53, 80_000} {
+				checkPortAndHTTPFormat(port, loggingv1.FormatKubeAPIAudit, `invalid port specified for HTTP receiver`)
+			}
+			checkPortAndHTTPFormat(8080, `no_such_format`, `invalid format specified for HTTP receiver`)
 			for _, port := range []int32{-1, 53, 80_000} {
-				checkPorts(port, loggingv1.FormatKubeAPIAudit, `invalid port specified for HTTP receiver`)
+				checkPortAndSyslogProtocol(port, "tcp", `invalid port specified for Syslog receiver`)
 			}
-			checkPorts(8080, `no_such_format`, `invalid format specified for HTTP receiver`)
-			checkReceiver(&loggingv1.ReceiverSpec{}, `ReceiverSpec must define an HTTP receiver`, map[string]bool{constants.VectorName: true})
+			checkPortAndSyslogProtocol(10514, "http", `invalid protocol specified for Syslog receiver`)
+			checkReceiverType("wrong-receiver", `invalid Type specified for receiver`)
+			checkReceiver(&loggingv1.ReceiverSpec{}, `invalid ReceiverTypeSpec specified for receiver`, map[string]bool{constants.VectorName: true})
 			checkReceiver(&loggingv1.ReceiverSpec{}, `ReceiverSpecs are only supported for the vector log collector`, map[string]bool{})
 		})
 
+		It("should fail validation if both HTTP and Syslog receivers are defined", func() {
```

Comment:
Does this matter if we are providing a "type" which is ... "use this one" ?

Reply:
No it does not, removed.